core/sqf/inc/cextdecs/cextdecs.h

```diff
@@ -289,6 +289,10 @@ _declspec(dllimport) short   DNUMOUT
 
 _declspec(dllimport) short   ENDTRANSACTION();
 
+//errStr is allocated if errlen is not zero. Caller must deallocate errStr.
+//Rest of functionality same as ENDTRANSACTION();
+_declspec(dllimport) short   ENDTRANSACTION_ERR(char *&errStr, int &errlen);
```

Comment:
Thanks @prashanth-vasudev for introducing another api to deallocate the error message. It is not  clear why is it missing  in the second time declaration in this file.